### DISCUSSION TITLE
DeepGram hook on non-public server

### DISCUSSION THREAD

**Discussion by arampacha** [deepgram/community#67] [General help] [answered]
I want to receive a callback from DeepGram on a server which is not publicly accessible. Can I get a range of IP addresses to whitelist for DeepGram?

**Comment by nikking-dg** [comment, marked as answer]
Hello and thanks for reaching out!

Because Deepgram leverages elastic cloud infrastructure, we don't maintain a range of IP addresses for callback requests.

To ensure callback validity, we recommend comparing the `request_id` returned in the request to the `request_id` in the callback. 

If you have a requirement to allow only allow specific IP addresses to access your internal network, another option would be to set up a proxy outside of your network (e.g. in a DMZ) that proxies Deepgram's callbacks.

**Comment by JamoCA** [comment]
I discovered this means that the DeepGram API endpoints don't use static IPs either and may change at any time.

JRE in a J2EE type installation typically has a default java.security setting that caches successful DNS host name resolutions forever.  This means that whenever elastic cloud IPs are reassigned, API requests will fail until the java service is manually restarted.

Please update API documentation to indicate this so that developers don't have to later find out the hard way.  Thanks.
